crates/sui-kvstore/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+mod bigtable;
+use anyhow::Result;
+use async_trait::async_trait;
+pub use bigtable::client::BigTableClient;
+use serde::{Deserialize, Serialize};
+use sui_types::base_types::SequenceNumber;
+use sui_types::digests::TransactionDigest;
+use sui_types::effects::{TransactionEffects, TransactionEvents};
+use sui_types::full_checkpoint_content::CheckpointData;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+use sui_types::object::Object;
+use sui_types::storage::ObjectKey;
+use sui_types::transaction::Transaction;
+
+#[async_trait]
+pub trait KeyValueStore {
+    async fn get_objects(&mut self, objects: &[ObjectKey]) -> Result<Vec<Object>>;
+    async fn get_transactions(
+        &mut self,
+        transactions: &[TransactionDigest],
+    ) -> Result<Vec<TransactionData>>;
+    async fn get_checkpoint(
+        &mut self,
+        sequence_number: SequenceNumber,
+    ) -> Result<Option<CheckpointData>>;
```

Comment:
I don't believe we want this to be CheckpointData but instead just (Summary, Sig, Contents)